### PR TITLE
fix(connector): file scan use correct path

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## What's changed and what's your intention?
https://github.com/risingwavelabs/risingwave/pull/19398 bring a bug that file_scan use incorrect path(With s3://bucket_name/ as prefix), so this pr fix this bug.

## Checklist

- [ ] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  My PR contains critical fixes that are necessary to be merged into the latest release. 

## Documentation

- [ ]  My PR needs documentation updates. 

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -10510,56 +10510,88 @@ name = "risingwave_batch"
...
  "either",
- "foyer",
  "futures",
  "futures-async-stream",
  "futures-util",
- "hashbrown 0.14.5",
- "hytra",
  "iceberg",
  "itertools 0.13.0",
+ "linkme",
```
Comment (wcy-fdu) [inline Cargo.lock:10522]: Should not change Cargo.lock, will remove.

**Comment by wcy-fdu** [comment]
todo: add e2e test

**Comment by wcy-fdu** [comment]
Tess passed.

**Review by zwang28 — APPROVED** [review]

File: src/connector/src/source/iceberg/parquet_file_handler.rs [hunk 2]
```diff
@@ -197,44 +198,39 @@ pub async fn list_s3_directory(
...
-        None => Ok(vec![]),
-    }
+                if is_parquet_schema_match_source_schema(arrow_data_type, rw_data_type) {
+                    Some(pos)
+                } else {
+                    None
+                }
+            })
+    })
+    .collect();
```
Comment (hzxa21) [inline src/connector/src/source/iceberg/parquet_file_handler.rs:232]: fmt?

**Review by hzxa21 — APPROVED** [review]
Rest LGTM. Should we add test cases for TVF file scan?

File: src/connector/src/source/iceberg/parquet_file_handler.rs (hunk 2 above)
Comment (wcy-fdu) [reply src/connector/src/source/iceberg/parquet_file_handler.rs:232]: Strange, how did these lines escape `cargo fmt` escape?

**Comment by wcy-fdu** [comment]
> Rest LGTM. Should we add test cases for TVF file scan?

I tried to add some test in CI, but found that currently tvf only support read file in s3, but not support read file in minio, see [definition of endpoint, it hard-coded into s3 instead of letting users pass it].(https://github.com/risingwavelabs/risingwave/blob/6f14e7937f3524a29ede0f7da4a3e71d8b65a3b9/src/connector/src/source/iceberg/parquet_file_handler.rs#L118-L121) .
Due to environmental issues, similar tests (parquet file source and file sink) are tested on minio.


I think we can add more backend such as 
```
file_scan(connector = minio)
file_scan(connector = gcs)
```
and then enhance ci test. This will take some time but this fix is a bit urgent, so I just test it locally and leave these in subsequent PRs.

**Pull request merged**